torch/distributed/tensor/_api.py

```diff
@@ -592,6 +613,16 @@ def placements(self) -> tuple[Placement, ...]:
         """
         return self._spec.placements
 
+    @property
+    def shard_order(self) -> Optional[tuple[tuple[int, ...], ...]]:
+        """
+        The shard_order attribute of this DTensor that describes the order of device
+        mesh dimensions each tensor dimension is sharded over.
```

Comment:
I think this needs more text, I don't think you can tell what the tuple of tuples is here directly.